chemdash.py

```diff
@@ -268,7 +254,7 @@ def get_columns_dropdown():
                 [dcc.Dropdown(
                         id='color_by_dropdown',
                         options=[
-                            {"label": i, "value": i} for i in get_columns_dropdown()
+                            {"label": i, "value": i} for i in get_columns_dropdown(df)
```

Comment:
Several functions now take the df as a parameter b/c it is no longer a global.  But is in scope for the whole app run.  

Managing this when very large gets more interesting.

Reply:
Hmmm, this is an interesting point, we might want to discuss how to handle dfs in an app like this.  Global  is nice for test case but not good overall.  Maybe we need some way to park the df when the user loads it and everytime they manipulate it.